src/dmangle.d

```diff
@@ -88,6 +94,9 @@ private immutable char[TMAX] mangleChar =
     //              X   // variadic T t...)
     //              Y   // variadic T t,...)
     //              Z   // not variadic, end of parameters
+    //              %   // Type backward reference
+    //              ?   // symbol backward reference
+    //              $   // identifier backward reference
```

Comment:
Agreed, I've been bit by `@` in symbol names, too.

`$` is currently used where identifiers have a different meaning, i.e. LName in the grammar.  Not sure it is easily replaceable. `?` would be similar.

`%` should be ok to replace, but I would suggest to use a single character encoding to not counter compression. If mangling is changed anyway, we could also move the deprecated complex types to double-letter encoding´to get a few free single letters.